cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/DBSink.java

```diff
@@ -141,11 +169,19 @@ private void setResultSetMetadata() throws Exception {
                                                                dbSinkConfig.columns, dbSinkConfig.tableName))
       ) {
         ResultSetMetaData resultSetMetadata = rs.getMetaData();
-        int columnCount = resultSetMetadata.getColumnCount();
-        columnTypes = new int[columnCount];
+        Map<String, Integer> columnToType = Maps.newHashMap();
         // JDBC driver column indices start with 1
-        for (int i = 0; i < columnCount; i++) {
-          columnTypes[i] = resultSetMetadata.getColumnType(i + 1);
+        for (int i = 0; i < rs.getMetaData().getColumnCount(); i++) {
+          String name = resultSetMetadata.getColumnName(i + 1);
+          int type = resultSetMetadata.getColumnType(i + 1);
+          columnToType.put(name, type);
+        }
+
+        columns = ImmutableList.copyOf(Splitter.on(",").split(dbSinkConfig.columns));
```

Comment:
can this (line 180-185) move outside the try-catch? Want to do as less as possible while the ``Connection is active.